//  File: quote.proto
//  Author: Tom Roeder <tmroeder@google.com>
//
//  Description: Protocol buffers for Tao Quote operations
//
//  Copyright (c) 2013, Google Inc.  All rights reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.



package tao;

message Quote {
  // the data to quote
  required bytes data = 1;

  required string hash_alg = 2;

  // the hash of the program making the statement
  required string hash = 3;

  // a sequence of quotes or other evidence showing that the signing key is correct
  repeated bytes evidence = 4;
}

message SignedQuote {
  required bytes serialized_quote = 1;
  required bytes signature = 2;
}